bravado/client.py

```diff
@@ -280,7 +280,7 @@ def construct_request(operation, request_options, **op_kwargs):
     """
     url = operation.swagger_spec.api_url.rstrip('/') + operation.path_name
     request = {
-        'method': operation.http_method.upper(),
+        'method': str(operation.http_method.upper()),
```

Comment:
str() does not convert to a binary string in Python 3 - as @macisamuele said a test for this would be great. `bytes()` is supported both in Python 2.7 and Python 3, but needs to be called differently depending on the Python version. You'll need to handle this a bit differently. What will do the trick is

    operation.http_method.upper().encode('ascii')

if the http_method is guaranteed to be str in Python 3 (it can be either str or unicode in Python 2).